x/emissions/proto/emissions/v9/events.proto

```diff
@@ -62,6 +67,322 @@ message EventNetworkInferences {
+  string reputer = 2;
+  string owner = 3;
+}
+
+message EventWorkerRegistered {
+  uint64 topic_id = 1;
+  string worker = 2;
+  string owner = 3;
+}
+
+message EventReputerUnregistered {
+  uint64 topic_id = 1;
+  string reputer = 2;
+}
+
+message EventWorkerUnregistered {
+  uint64 topic_id = 1;
+  string worker = 2;
+}
+
+message EventFundTopic {
+  uint64 topic_id = 1;
+  string funder = 2;
+  string amount = 3 [
+    (cosmos_proto.scalar) = "cosmos.Int",
+    (gogoproto.customtype) = "cosmossdk.io/math.Int",
+    (gogoproto.nullable) = false,
+    (amino.dont_omitempty) = true
+  ];
+}
+
+message EventParamsSet {
+  emissions.v8.Params params = 1;
+}
+
+message EventWhitelistAdminAdded {
+  string admin = 1;
+}
+
+message EventWhitelistAdminRemoved {
+  string admin = 1;
+}
+
+message EventGlobalWhitelistAdded {
+  string address = 1;
+}
+
+message EventGlobalWhitelistRemoved {
+  string address = 1;
+}
+
+message EventGlobalWorkerWhitelistAdded {
+  string address = 1;
+}
+
+message EventGlobalWorkerWhitelistRemoved {
+  string address = 1;
+}
+
+message EventGlobalReputerWhitelistAdded {
+  string address = 1;
+}
+
+message EventGlobalReputerWhitelistRemoved {
+  string address = 1;
+}
+
+message EventGlobalAdminWhitelistAdded {
+  string address = 1;
+}
+
+message EventGlobalAdminWhitelistRemoved {
+  string address = 1;
+}
+
+message EventGlobalWorkerWhitelistBulkAdded {
+  repeated string addresses = 1;
+}
+
+message EventGlobalWorkerWhitelistBulkRemoved {
+  repeated string addresses = 1;
+}
+
+message EventGlobalReputerWhitelistBulkAdded {
+  repeated string addresses = 1;
+}
+
+message EventGlobalReputerWhitelistBulkRemoved {
+  repeated string addresses = 1;
+}
+
+message EventGlobalAdminWhitelistBulkAdded {
+  repeated string addresses = 1;
+}
+
+message EventGlobalAdminWhitelistBulkRemoved {
+  repeated string addresses = 1;
+}
+
+message EventTopicWorkerWhitelistEnabled {
+  uint64 topic_id = 1;
+}
+
+message EventTopicWorkerWhitelistDisabled {
+  uint64 topic_id = 1;
+}
+
+message EventTopicReputerWhitelistEnabled {
+  uint64 topic_id = 1;
+}
+
+message EventTopicReputerWhitelistDisabled {
+  uint64 topic_id = 1;
+}
+
+message EventTopicCreatorWhitelistAdded {
+  uint64 topic_id = 1;
+  string address = 2;
+}
+
+message EventTopicCreatorWhitelistRemoved {
+  uint64 topic_id = 1;
+  string address = 2;
+}
+
+message EventTopicWorkerWhitelistAdded {
+  uint64 topic_id = 1;
+  string address = 2;
+}
+
+message EventTopicWorkerWhitelistRemoved {
+  uint64 topic_id = 1;
+  string address = 2;
+}
+
+message EventTopicReputerWhitelistAdded {
+  uint64 topic_id = 1;
+  string address = 2;
+}
+
+message EventTopicReputerWhitelistRemoved {
+  uint64 topic_id = 1;
+  string address = 2;
+}
+
+message EventTopicAdminWhitelistAdded {
+  uint64 topic_id = 1;
+  string address = 2;
+}
+
+message EventTopicAdminWhitelistRemoved {
+  uint64 topic_id = 1;
+  string address = 2;
+}
+
+message EventTopicAdminWhitelistBulkAdded {
```

Comment:
I think we should have a single event associated to a specific state mutation, bulk operations just run the same mutation multiple times which should emit multiple events, not a different one imho

Reply:
To be honest I don't think having bulk msg is a good practice as we can just put multiple message in a tx.
Maybe a compromise can be found with a single event type having a repeated field.